great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -2583,7 +2583,7 @@ def _sqlalchemy_map_condition_index(
     metric_value_kwargs: Dict,
     metrics: Dict[str, Any],
     **kwargs,
-) -> List[Dict[str, Any]]:
+) -> List[Dict[str, Any]] | None:
```

Comment:
Nit - if we're using `|`, could we change this to `list[dict[str, Any]] | None`?